src/lib/salad/hll.c

```diff
@@ -0,0 +1,223 @@
+#include "hll.h"
+#include "hll_empirical.h"
+
+#include <trivia/util.h>
+
+#include <stddef.h>
+#include <math.h>
+#include <assert.h>
+
+/* 6 bits are needed to store the number of leading zeros of 64 bit hash */
+#define HLL_REGISTER_BITS  6
```

Comment:
Would be better to use enum or const uint_..

Reply:
I'm afraid `const` won't work because `const` variables are not considered compile-time evaluable by C standard. But `enum` seems OK.

Reply:
I don't quite understand what's going on here: isn't `HLL_REGISTER_BITS` supposed to be a power of 2?